cmd/completion/completion.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Silvio Böhler
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package completion
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func CreateCmd(rootCmd *cobra.Command) *cobra.Command {
+	c := &cobra.Command{
+		Use:   "completion",
+		Short: "output shell completion code",
+		Long: "Generate shell completion code for `bash` or `zsh` shells.",
+
+		Args: cobra.ExactValidArgs(1),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			switch args[0] {
+			case `bash`:
+				rootCmd.GenBashCompletion(os.Stdout)
+			case `zsh`:
+				rootCmd.GenZshCompletion(os.Stdout)
+				io.WriteString(os.Stdout, "\ncompdef _knut knut\n")
```

Comment:
What does this line do? I don't see it in https://github.com/spf13/cobra/blob/master/shell_completions.md

Reply:
That is an interesting line it sources the completion definition file, and tells ZSH what command the completions are for [1]. Cobra does not include this line in the generated output [2]. 

[1] https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org#telling-zsh-which-function-to-use-for-completing-a-command

[2] https://github.com/helm/helm/blob/master/cmd/helm/completion.go#L167

Reply:
Thanks!